File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-1.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        vector<bool> rows(matrix.size(), false);
        vector<bool> col(matrix[0].size(), false);
        for(int i = 0; i<matrix.size(); i++)
        {
            for(int j = 0; j<matrix[0].size(); j++)
            {
                if(matrix[i][j]==0)
                {
                    rows[i] = true;
                    col[j] = true;
                }
            }
        }
        for(int k = 0; k<matrix.size(); k++)
        {
            for(int l = 0; l<matrix[0].size(); l++)
            {
                if(rows[k] || col[l])
                {
                    cout << k << " " << l << "\n";
                    matrix[k][l] = 0;
                }
            }
        }
    }
};

File: Data Structures & Algorithms/spiral-matrix/submission-1.java
class Solution {
    public List<Integer> spiralOrder(int[][] matrix) {
        int top = 0;
        int right = matrix[0].length-1;
        int bottom = matrix.length-1;
        int left = 0;
        List<Integer> ans = new ArrayList<>();
        while(top<=bottom && left<=right)
        {
        for(int i = left; i<=right; i++)
        {
            ans.add(matrix[top][i]);
        }
        top++;
        for(int j = top; j<=bottom; j++)
        {
            ans.add(matrix[j][right]);
        }
        right--;
        if(top<=bottom)
        {
        for(int k = right; k>=left; k--)
        {
            ans.add(matrix[bottom][k]);
        }
        bottom--;
        }
        if(left<=right)
        {
        for(int l = bottom; l>=top; l--)
        {
            ans.add(matrix[l][left]);
        }
        left++;
        }
    }
            return ans;
    }
}
